packages/flutter/lib/src/material/app.dart

```diff
@@ -403,45 +442,80 @@ class _MaterialAppState extends State<MaterialApp> {
-    Widget result = AnimatedTheme(
-      data: theme,
-      isMaterialAppTheme: true,
-      child: WidgetsApp(
-        key: GlobalObjectKey(this),
-        navigatorKey: widget.navigatorKey,
-        navigatorObservers: _navigatorObservers,
+    Widget result = WidgetsApp(
+      key: GlobalObjectKey(this),
+      navigatorKey: widget.navigatorKey,
+      navigatorObservers: _navigatorObservers,
         pageRouteBuilder: <T>(RouteSettings settings, WidgetBuilder builder) =>
-          MaterialPageRoute<T>(settings: settings, builder: builder),
-        home: widget.home,
-        routes: widget.routes,
-        initialRoute: widget.initialRoute,
-        onGenerateRoute: widget.onGenerateRoute,
-        onUnknownRoute: widget.onUnknownRoute,
-        builder: widget.builder,
-        title: widget.title,
-        onGenerateTitle: widget.onGenerateTitle,
-        textStyle: _errorTextStyle,
-        // blue is the primary color of the default theme
-        color: widget.color ?? theme?.primaryColor ?? Colors.blue,
-        locale: widget.locale,
-        localizationsDelegates: _localizationsDelegates,
-        localeResolutionCallback: widget.localeResolutionCallback,
-        localeListResolutionCallback: widget.localeListResolutionCallback,
-        supportedLocales: widget.supportedLocales,
-        showPerformanceOverlay: widget.showPerformanceOverlay,
-        checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
-        checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
-        showSemanticsDebugger: widget.showSemanticsDebugger,
-        debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
-        inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
-          return FloatingActionButton(
-            child: const Icon(Icons.search),
-            onPressed: onPressed,
-            mini: true,
-          );
-        },
-      ),
+            MaterialPageRoute<T>(settings: settings, builder: builder),
+      home: widget.home,
+      routes: widget.routes,
+      initialRoute: widget.initialRoute,
+      onGenerateRoute: widget.onGenerateRoute,
+      onUnknownRoute: widget.onUnknownRoute,
+      builder: (BuildContext context, Widget child) {
+        // Use a light theme, dark theme, or fallback theme.
+        ThemeData theme;
+        final ui.Brightness platformBrightness = MediaQuery.platformBrightnessOf(context);
+        if (platformBrightness == ui.Brightness.dark && widget.darkTheme != null) {
+          theme = widget.darkTheme;
+        } else if (widget.theme != null) {
+          theme = widget.theme;
+        } else {
+          theme = ThemeData.fallback();
+        }
+
+        return AnimatedTheme(
+          data: theme,
+          isMaterialAppTheme: true,
+          child: widget.builder != null
+              ? Builder(
+                  builder: (BuildContext context) {
+                    // Why are we surrounding a builder with a builder?
```

Comment:
This seems like a generic problem with AnimatedTheme. Maybe that's where the extra BuildContext should be inserted. 

Could be fixed in a separate PR.

Reply:
Yeah, I'll let that be investigated on its own.  Seems like it could be a deep investigation with regard to performance implications and also backwards compatibility.